From 47bc5a2d8edf8087986aca82151f16f48497402d Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Tue, 25 Nov 2014 11:19:08 +0800
Subject: [PATCH 200/210] disable platform clock by default

By default, all the platform clock should be off post boot, the one
who use its clock should enable or disable itself within its driver.

Change-Id: Ifb44a6f83d3bd223914df587b55b959eec2705fc
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
---
 drivers/misc/vlv2_plat_clock.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/misc/vlv2_plat_clock.c b/drivers/misc/vlv2_plat_clock.c
index b3821f9..e75061a 100644
--- a/drivers/misc/vlv2_plat_clock.c
+++ b/drivers/misc/vlv2_plat_clock.c
@@ -188,12 +188,18 @@ EXPORT_SYMBOL_GPL(vlv2_plat_get_clock_status);
 
 static int vlv2_plat_clk_probe(struct platform_device *pdev)
 {
+	int i = 0;
+
 	pmc_base = ioremap_nocache(VLV2_PMC_CLK_BASE_ADDRESS, PMC_MAP_SIZE);
 	if (!pmc_base) {
 		dev_err(&pdev->dev, "I/O memory remapping failed\n");
 		return -ENOMEM;
 	}
 
+	/* Initialize all clocks as disabled */
+	for (i = 0; i < MAX_CLK_COUNT; i++)
+		vlv2_plat_configure_clock(i, CLK_CONFG_FORCE_OFF);
+
 	dev_info(&pdev->dev, "vlv2_plat_clk initialized\n");
 	return 0;
 }
-- 
1.9.1

